lib/winrm-fs/file_manager.rb

```diff
@@ -59,16 +59,43 @@ def delete(path)
       # Downloads the specified remote file to the specified local path
       # @param [String] The full path on the remote machine
       # @param [String] The full path to write the file to locally
-      def download(remote_path, local_path, first = true)
+      # rubocop:disable Metrics/MethodLength
+      def download(remote_path, local_path, first = true, chunk_size = 1024 * 1024, index = 0)
         @logger.debug("downloading: #{remote_path} -> #{local_path}")
-        script = WinRM::FS::Scripts.render('download', path: remote_path)
-        output = @connection.shell(:powershell) { |e| e.run(script) }
-        contents = output.stdout.gsub('\n\r', '')
-        return false if output.exitcode != 0
-        download_dir(remote_path, local_path, first) if contents.empty?
-        IO.binwrite(local_path, Base64.decode64(contents)) unless contents.empty?
+        output = _output_from_file(remote_path, chunk_size, index)
+        return download_dir(remote_path, local_path, first) if output.exitcode == 2
+
+        return false if output.exitcode >= 1
+
+        File.open(local_path, 'wb') do |fd|
+          out = _write_file(fd, output)
+          index += out.length
+          until out.empty?
+            output = _output_from_file(remote_path, chunk_size, index)
+            return false if output.exitcode >= 1
+
+            out = _write_file(fd, output)
+            index += out.length
+          end
+        end
         true
       end
+      # rubocop:enable Metrics/MethodLength
+
+      def _output_from_file(remote_path, chunk_size = 1024 * 1024, index = 0)
```

Comment:
you should not need default vals here since all callers pass in explicit values.